extensions/big-data-cluster/src/bigDataCluster/controller/clusterControllerApi.ts

```diff
@@ -173,14 +173,14 @@ export class ClusterController implements IClusterController {
 		}
 	}
 
-	public async getKnoxUsername(sqlLogin: string): Promise<string> {
+	public async getKnoxUsername(defaultUsername: string): Promise<string> {
 		// This all is necessary because prior to CU5 BDC deployments all had the same default username for
 		// accessing the Knox gateway. But in the allowRunAsRoot setting was added and defaulted to false - so
 		// if that exists and is false then we use the username instead.
 		// Note that the SQL username may not necessarily be correct here either - but currently this is what
 		// we're requiring to run Notebooks in a BDC
 		const config = await this.getClusterConfig();
-		return config.spec?.spec?.security?.allowRunAsRoot === false ? sqlLogin : DEFAULT_KNOX_USERNAME;
+		return config.spec?.spec?.security?.allowRunAsRoot === false ? defaultUsername : DEFAULT_KNOX_USERNAME;
```

Comment:
either defaultUsername is a not a good name or DEFAULT_KNOX_USERNAME is not a good choice for the name. The way it currently reads is that we are returning one of two defaults for the KnoxUsernames which is a little odd.